client/packages/common/src/utils/BarcodeScannerContext.tsx

```diff
@@ -184,20 +187,24 @@ export const BarcodeScannerProvider: FC<PropsWithChildrenOnly> = ({
     }
 
     if (hasNativeBarcodeScanner) {
-      await BarcodeScanner.stopScan({ resolveScan: true });
-      await BarcodeScanner.showBackground();
+      await BarcodeScannerPlugin.stopScan({ resolveScan: true });
+      await BarcodeScannerPlugin.showBackground();
     }
   };
 
+  // calling this outside of a useEffect so that it will detect when a new scanner is added
```

Comment:
hmm `will detect when a new scanner is added` do you mean when this context re renders ? When device is found in ElectronSettings, how do we know it this Context will re render and run barcodeScannerDevice() ?

I think it should be in use effect (to happen at the start) and this method is exposed in state so that it can be set in ElectronSettings

Reply:
ah. yes - that's how I had it, and then realised it wouldn't re-render.
Adding in a call to ` useBarcodeScannerContext` is the missing link. thanks!